src/encoder.rs

```diff
@@ -2362,7 +2367,7 @@ fn encode_partition_topdown(seq: &Sequence, fi: &FrameInvariants, fs: &mut Frame
     if must_split {
         // Oversized blocks are split automatically
         partition = PartitionType::PARTITION_SPLIT;
```

Comment:
When must_split is true, there may be cases where it is ok to use a horizontal or a vertical split, which may be more effective

Reply:
Indeed, I will add a patch for it and report the gain by it.

Reply:
After further checking and trying some implement, I think this place is not where it considers rectangular partitions with topdown. Because, rect partition is the leaf nodes, so for ex, once 64x64 is given then say it straddle on the right frame border thus wants to use 32x64 but that is not possible if only 8 pixels are outside the right frame border.
Instead, as in current code intends, once 16x16 square block is given to topdown_partition_rdo(), then it calls rdo_partition_decision() can pick 8x16 partition, if it beats 8x8s. HOWEVER, that seems not happening now, unfortunately, which I should fix.

Reply:
Also, by av1 spec, the small rec partition size, like 8x4 or 4x8 are encoded by way of series of PARTITION_SPLITS to reach its next larger sqr size, i.e. 8x8 then finally encode PARTITION_HORZ or PARTITION_VERT.